### DISCUSSION TITLE
I'm having issues with the message version bump.

### DISCUSSION THREAD

**Discussion by tymesink** [GitTools/GitVersion#3933] [Q&A] [answered]
I am using azure classic build pipeline with the gitVersion/setup and gitVersion/execute task. Here is my gitVersion config:

> mode: ContinuousDeployment
> tag-prefix: '[vV]'
> commit-message-incrementing: Enabled
> major-version-bump-message: '\+semver:\s?(breaking|major)'
> minor-version-bump-message: '\+semver:\s?(feature|minor)'
> patch-version-bump-message: '\+semver:\s?(fix|bugfix|hotfix|patch)'
> no-bump-message: '\+semver:\s?(none|skip)'
> branches:
>   main:
>     regex: ^main$
>     mode: ContinuousDelivery
>     tag: ''
>     increment: Patch
>     is-source-branch-for: ['hotfix']
>     source-branches: ['develop', 'hotfix']
>     prevent-increment-of-merged-branch-version: true
>     track-merge-target: false
>     tracks-release-branches: false
>     is-release-branch: false
>     is-mainline: true
>   develop:
>     regex: ^dev(elop)?(ment)?$
>     mode: ContinuousDelivery    
>     tag: 'dev'
>     increment: Minor
>     is-source-branch-for: ['main']
>     source-branches: []
>     prevent-increment-of-merged-branch-version: false
>     tracks-release-branches: true
>     track-merge-target: true
>     is-release-branch: false
>     is-mainline: false
>   hotfix:
>     regex: ^hotfix$
>     mode: ContinuousDelivery
>     tag: 'hotfix'
>     increment: Patch
>     source-branches: ['main']
>     is-source-branch-for: ['main']
>     prevent-increment-of-merged-branch-version: false
>     tracks-release-branches: false
>     track-merge-target: false
>     is-release-branch: false
>     is-mainline: false
> merge-message-formats: {}
> ignore:
>   sha: []

GitVersion is behaving as expected producing the propriate versions as specified by the config. However, when I try to do a patch-version-bump-message, nothing happens in the develop branch by adding the bump message in the PR and GitVersion just increases the commits on the prepatch tag.

For instance if the previous version was 1.4.0-dev.3 and then I created a branch from develop, made some changes, commit, push and create a PR. In the PR I added +semver: patch and then approved the PR. GitVersion calculates the next version to be 1.4.0-dev.4. 

I outputted the git log for the last few commits:
`commit b7463218a0104810a46b4279e45f6741ca4060eb
Author: JohnnyDoe  <js@test.com>
Date:   Mon Feb 26 20:30:57 2024 +0000

    Merged PR 83: Another Feature
    
    [DO NOT DELETE] +semver: patch
    Anotoher change

commit bf180405399e5dd7fe010c2b45c6971efec68f76
Author: JohnnyDoe <js@test.com>
Date:   Mon Feb 26 17:59:16 2024 +0000

    Merged PR 82: Added new class
    
    - Added new class
    [DO NOT DELETE] +semver: patch

I am assuming I am doing something wrong, or the config is wrong, or my expectation is wrong. How would I manually bump the version using Visual Studio / Azure Devops PR workflow (without going to the git cmd/bash)?

Here is the logs from the gitVersion task:

> 2024-02-26T20:36:39.4047705Z ==============================================================================
> 2024-02-26T20:36:39.4048198Z Task         : Execute GitVersion Task
> 2024-02-26T20:36:39.4048277Z Description  : Easy Semantic Versioning (http://semver.org) for projects using Git
> 2024-02-26T20:36:39.4048399Z Version      : 0.11.0
> 2024-02-26T20:36:39.4048457Z Author       : GitTools Contributors
> 2024-02-26T20:36:39.4048527Z Help         : See the [documentation](https://gitversion.net/docs/) for help
> 2024-02-26T20:36:39.4048616Z ==============================================================================
> 2024-02-26T20:36:39.8261581Z Agent: 'Azure Pipelines'
> 2024-02-26T20:36:39.8302986Z Command: dotnet-gitversion D:/a/1/s /output json /output buildserver /config D:\a\1\s\Build\gitversion.config.yml
> 2024-02-26T20:36:39.8343263Z [command]C:\hostedtoolcache\windows\GitVersion.Tool\5.12.0\x64\dotnet-gitversion.exe D:/a/1/s /output json /output buildserver /config D:\a\1\s\Build\gitversion.config.yml
> 2024-02-26T20:36:42.2803853Z INFO [02/26/24 20:36:40:41] Working directory: D:/a/1/s
> 2024-02-26T20:36:42.2804150Z INFO [02/26/24 20:36:40:67] Project root is: D:\a\1\s\
> 2024-02-26T20:36:42.2804331Z INFO [02/26/24 20:36:40:67] DotGit directory is: D:\a\1\s\.git
> 2024-02-26T20:36:42.2804548Z INFO [02/26/24 20:36:40:67] Branch from build environment: refs/heads/develop
> 2024-02-26T20:36:42.2808954Z INFO [02/26/24 20:36:40:67] Begin: Normalizing git directory for branch 'refs/heads/develop'
> 2024-02-26T20:36:42.2809724Z   INFO [02/26/24 20:36:40:71] One remote found (origin -> 'https://foo@dev.azure.com/foo/foo-projects/_git/GitVersionTest').
> 2024-02-26T20:36:42.2810080Z   INFO [02/26/24 20:36:40:71] Skipping fetching, if GitVersion does not calculate your version as expected you might need to allow fetching or use dynamic repositories
> 2024-02-26T20:36:42.2810362Z   INFO [02/26/24 20:36:40:72] Creating local branch develop
> 2024-02-26T20:36:42.2812233Z   INFO [02/26/24 20:36:40:78] Creating local branch from remote tracking 'refs/remotes/origin/hotfix'.
> 2024-02-26T20:36:42.2812681Z   INFO [02/26/24 20:36:40:79] Creating local branch from remote tracking 'refs/remotes/origin/main'.
> 2024-02-26T20:36:42.2813030Z   INFO [02/26/24 20:36:40:80] HEAD points at branch 'refs/heads/develop'.
> 2024-02-26T20:36:42.2813421Z   INFO [02/26/24 20:36:40:81] End: Normalizing git directory for branch 'refs/heads/develop' (Took: 133.87ms)
> 2024-02-26T20:36:42.2813657Z   INFO [02/26/24 20:36:40:82] Begin: Loading version variables from disk cache
> 2024-02-26T20:36:42.2813927Z   INFO [02/26/24 20:36:40:83] Cache file D:\a\1\s\.git\gitversion_cache\D0567FA236275D48AF416F36C52732A74116BDA9.yml not found.
> 2024-02-26T20:36:42.2814182Z   INFO [02/26/24 20:36:40:83] End: Loading version variables from disk cache (Took: 0.95ms)
> 2024-02-26T20:36:42.2814410Z   INFO [02/26/24 20:36:40:83] Using latest commit on specified branch
> 2024-02-26T20:36:42.2814629Z   INFO [02/26/24 20:36:41:03] Running against branch: develop (b746321 Merged PR 83: Another Feature)
> 2024-02-26T20:36:42.2814861Z   INFO [02/26/24 20:36:41:04] Begin: Calculating base versions
> 2024-02-26T20:36:42.2815116Z     INFO [02/26/24 20:36:41:07] Fallback base version: 0.1.0 with commit count source f45083a8feb5e5c8a0cbf007bf1b460c661a1beb
> 2024-02-26T20:36:42.2815609Z     INFO [02/26/24 20:36:41:17] Git tag 'v1.4.0-dev.4': 1.4.0-dev.4 with commit count source bf180405399e5dd7fe010c2b45c6971efec68f76
> 2024-02-26T20:36:42.2816114Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.4.0-dev.3': 1.4.0-dev.3 with commit count source 603410ee39f066ce33238b541f9185e24b993b17
> 2024-02-26T20:36:42.2819136Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.4.0-dev.2': 1.4.0-dev.2 with commit count source b0de769897749333d2ad369bcf5383f434df27e9
> 2024-02-26T20:36:42.2821485Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.4.0-dev.1': 1.4.0-dev.1 with commit count source 978fd5b2d05c64f06f28c3d9ef7a4413e4d27105
> 2024-02-26T20:36:42.2823219Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.3.0-dev.1': 1.3.0-dev.1 with commit count source 7ebeda1911f7bb7ef3377ed29ab835212afd8afe
> 2024-02-26T20:36:42.2823884Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.2.0-dev.2': 1.2.0-dev.2 with commit count source 8795e0407b82e4c13286775cddd24b3b5e27287e
> 2024-02-26T20:36:42.2824346Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.2.0-dev.1': 1.2.0-dev.1 with commit count source 973f7d1365c6e779f5ee94798bff36fd955b00a6
> 2024-02-26T20:36:42.2824805Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.1.1': 1.1.1 with commit count source a066704b4fd8025bccfeb2ff2def56c9f90b39dd
> 2024-02-26T20:36:42.2825276Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.0.2-hotfix.1': 1.0.2-hotfix.1 with commit count source 55ba1d828c652bd7e4865d47910c7236ee69ce86
> 2024-02-26T20:36:42.2825742Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.1.0': 1.1.0 with commit count source 9d2abbb68dbbf303ba8ae22c457e89d47816168c
> 2024-02-26T20:36:42.2826187Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.1.0-dev.2': 1.1.0-dev.2 with commit count source 4efa69cd06279d0fa11fe4f289d8352f9d943cfd
> 2024-02-26T20:36:42.2826892Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.1.0-dev.1': 1.1.0-dev.1 with commit count source b057a461400e7f138172bf7b73fbec6b2a10d920
> 2024-02-26T20:36:42.2827350Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.0.1': 1.0.1 with commit count source 8da13ce2243ead244cdbc7437a854065171f8f5d
> 2024-02-26T20:36:42.2827776Z     INFO [02/26/24 20:36:41:18] Git tag 'v1.0.0': 1.0.0 with commit count source 5818044f9b880ed2439eb3db1e96a3e693797c97
> 2024-02-26T20:36:42.2828217Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.3.0': 1.3.0 with commit count source 0f96824d8826b9d9d1a07913defe12e005338f06
> 2024-02-26T20:36:42.2828657Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.3.0-dev.1': 1.3.0-dev.1 with commit count source 7ebeda1911f7bb7ef3377ed29ab835212afd8afe
> 2024-02-26T20:36:42.2829111Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.2.0': 1.2.0 with commit count source 9ac1346adc889af98207dbeec84efd48fb42fab9
> 2024-02-26T20:36:42.2829550Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.2.0-dev.2': 1.2.0-dev.2 with commit count source 8795e0407b82e4c13286775cddd24b3b5e27287e
> 2024-02-26T20:36:42.2830023Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.2.0-dev.1': 1.2.0-dev.1 with commit count source 973f7d1365c6e779f5ee94798bff36fd955b00a6
> 2024-02-26T20:36:42.2830456Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.1.1': 1.1.1 with commit count source a066704b4fd8025bccfeb2ff2def56c9f90b39dd
> 2024-02-26T20:36:42.2830931Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.0.2-hotfix.1': 1.0.2-hotfix.1 with commit count source 55ba1d828c652bd7e4865d47910c7236ee69ce86
> 2024-02-26T20:36:42.2831372Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.1.0': 1.1.0 with commit count source 9d2abbb68dbbf303ba8ae22c457e89d47816168c
> 2024-02-26T20:36:42.2831831Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.1.0-dev.2': 1.1.0-dev.2 with commit count source 4efa69cd06279d0fa11fe4f289d8352f9d943cfd
> 2024-02-26T20:36:42.2832282Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.1.0-dev.1': 1.1.0-dev.1 with commit count source b057a461400e7f138172bf7b73fbec6b2a10d920
> 2024-02-26T20:36:42.2832734Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.0.1': 1.0.1 with commit count source 8da13ce2243ead244cdbc7437a854065171f8f5d
> 2024-02-26T20:36:42.2833176Z     INFO [02/26/24 20:36:41:19] Git tag 'v1.0.0': 1.0.0 with commit count source 5818044f9b880ed2439eb3db1e96a3e693797c97
> 2024-02-26T20:36:42.2833661Z     INFO [02/26/24 20:36:41:37] Found multiple base versions which will produce the same SemVer (1.4.0), taking oldest source for commit counting (Git tag 'v1.3.0')
> 2024-02-26T20:36:42.2834311Z     INFO [02/26/24 20:36:41:37] Base version used: Git tag 'v1.3.0': 1.3.0 with commit count source 0f96824d8826b9d9d1a07913defe12e005338f06
> 2024-02-26T20:36:42.2834570Z   INFO [02/26/24 20:36:41:37] End: Calculating base versions (Took: 331.08ms)
> 2024-02-26T20:36:42.2835054Z   INFO [02/26/24 20:36:41:37] 5 commits found between 0f96824 Merged PR 77: Merged PR 76: Feature-DoWork1 and b746321 Merged PR 83: Another Feature
> 2024-02-26T20:36:42.2835550Z   INFO [02/26/24 20:36:41:38] 5 commits found between 0f96824 Merged PR 77: Merged PR 76: Feature-DoWork1 and b746321 Merged PR 83: Another Feature
> 2024-02-26T20:36:42.2838392Z   INFO [02/26/24 20:36:41:38] Begin: Getting version tags from branch 'refs/heads/develop'.
> 2024-02-26T20:36:42.2838808Z   INFO [02/26/24 20:36:41:39] End: Getting version tags from branch 'refs/heads/develop'. (Took: 10.94ms)
> 2024-02-26T20:36:42.2839050Z   INFO [02/26/24 20:36:41:40] Begin: Creating dictionary
> 2024-02-26T20:36:42.2839245Z   INFO [02/26/24 20:36:41:40] End: Creating dictionary (Took: 5.70ms)
> 2024-02-26T20:36:42.2839523Z   INFO [02/26/24 20:36:41:40] Begin: Storing version variables to cache file D:\a\1\s\.git\gitversion_cache\D0567FA236275D48AF416F36C52732A74116BDA9.yml
> 2024-02-26T20:36:42.2839852Z   INFO [02/26/24 20:36:41:44] End: Storing version variables to cache file D:\a\1\s\.git\gitversion_cache\D0567FA236275D48AF416F36C52732A74116BDA9.yml (Took: 38.99ms)
> 2024-02-26T20:36:42.2840227Z Executing GenerateSetVersionMessage for 'AzurePipelines'.
> 2024-02-26T20:36:42.3222885Z Executing GenerateBuildLogOutput for 'AzurePipelines'.
> 2024-02-26T20:36:42.3298995Z {
> 2024-02-26T20:36:42.3299153Z   "Major": 1,
> 2024-02-26T20:36:42.3299260Z   "Minor": 4,
> 2024-02-26T20:36:42.3299417Z   "Patch": 0,
> 2024-02-26T20:36:42.3299534Z   "PreReleaseTag": "dev.5",
> 2024-02-26T20:36:42.3299851Z   "PreReleaseTagWithDash": "-dev.5",
> 2024-02-26T20:36:42.3301285Z   "PreReleaseLabel": "dev",
> 2024-02-26T20:36:42.3301597Z   "PreReleaseLabelWithDash": "-dev",
> 2024-02-26T20:36:42.3301841Z   "PreReleaseNumber": 5,
> 2024-02-26T20:36:42.3301974Z   "WeightedPreReleaseNumber": 5,
> 2024-02-26T20:36:42.3302362Z   "BuildMetaData": 5,
> 2024-02-26T20:36:42.3302484Z   "BuildMetaDataPadded": "0005",
> 2024-02-26T20:36:42.3302709Z   "FullBuildMetaData": "5.Branch.develop.Sha.b7463218a0104810a46b4279e45f6741ca4060eb",
> 2024-02-26T20:36:42.3302879Z   "MajorMinorPatch": "1.4.0",
> 2024-02-26T20:36:42.3303298Z   "SemVer": "1.4.0-dev.5",
> 2024-02-26T20:36:42.3303505Z   "LegacySemVer": "1.4.0-dev5",
> 2024-02-26T20:36:42.3303774Z   "LegacySemVerPadded": "1.4.0-dev0005",
> 2024-02-26T20:36:42.3303910Z   "AssemblySemVer": "1.4.0.0",
> 2024-02-26T20:36:42.3304091Z   "AssemblySemFileVer": "1.4.0.0",
> 2024-02-26T20:36:42.3304296Z   "FullSemVer": "1.4.0-dev.5+5",
> 2024-02-26T20:36:42.3304610Z   "InformationalVersion": "1.4.0-dev.5+5.Branch.develop.Sha.b7463218a0104810a46b4279e45f6741ca4060eb",
> 2024-02-26T20:36:42.3306147Z   "BranchName": "develop",
> 2024-02-26T20:36:42.3306293Z   "EscapedBranchName": "develop",
> 2024-02-26T20:36:42.3306542Z   "Sha": "b7463218a0104810a46b4279e45f6741ca4060eb",
> 2024-02-26T20:36:42.3306683Z   "ShortSha": "b746321",
> 2024-02-26T20:36:42.3307013Z   "NuGetVersionV2": "1.4.0-dev0005",
> 2024-02-26T20:36:42.3307243Z   "NuGetVersion": "1.4.0-dev0005",
> 2024-02-26T20:36:42.3307496Z   "NuGetPreReleaseTagV2": "dev0005",
> 2024-02-26T20:36:42.3307634Z   "NuGetPreReleaseTag": "dev0005",
> 2024-02-26T20:36:42.3307857Z   "VersionSourceSha": "0f96824d8826b9d9d1a07913defe12e005338f06",
> 2024-02-26T20:36:42.3308205Z   "CommitsSinceVersionSource": 5,
> 2024-02-26T20:36:42.3308413Z   "CommitsSinceVersionSourcePadded": "0005",
> 2024-02-26T20:36:42.3308554Z   "UncommittedChanges": 0,
> 2024-02-26T20:36:42.3308837Z   "CommitDate": "2024-02-26"
> 2024-02-26T20:36:42.3311979Z }
> 2024-02-26T20:36:42.3314503Z   INFO [02/26/24 20:36:41:73] Done writing 
> 2024-02-26T20:36:42.3794030Z ##[section]Async Command Start: Update Build Number
> 2024-02-26T20:36:42.5667399Z Update build number to 1.4.0-dev.5+5 for build 253
> 2024-02-26T20:36:42.5667938Z ##[section]Async Command End: Update Build Number
> 2024-02-26T20:36:42.5669048Z ##[section]Finishing: gitversion/execute

**Comment by HHobeck** [comment, marked as answer]
The bump message feature like it is implemented in the actual version of git-version is there to bump the increment to a higher value for the calculation of the next version. If lets say the base version is 1.0.0 then the following matrix applies:

A commit without a bump message yields:
- on branches with increment none to 1.0.0
- on branches with increment patch to 1.0.1
- on branches with increment minor to 1.1.0
- on branches with increment major to 2.0.0

A commit with semver+: patch message yields
- on branches with increment none to 1.0.1
- on branches increment patch to 1.0.1
- on branches increment minor to 1.1.0
- on branches increment major to 2.0.0

A commit with semver+: minor message yields
- on branches with increment none to 1.1.0
- on branches with increment patch to 1.1.0
- on branches with increment minor to 1.1.0
- on branches with increment major to 2.0.0

A commit with semver+: major message yields
- on branches with increment none to 2.0.0
- on branches with increment patch to 2.0.0
- on branches with increment minor to 2.0.0
- on branches with increment major to 2.0.0

You need to think about the use case of using the bump message feature. The motivation is more dependent on what changes your commit includes from the development perspective. When you in the role of a developer don't care about what the target branch might be and think: "Oh it's not just a minor change it’s a huge change which should yield to a major version increment because I have updated the API with a breaking change."

**Reply by tymesink** [reply]
From your reply...

> A commit with semver+: patch message yields
> 
> on branches with increment none to 1.0.1
> on branches increment patch to 1.0.1
> _**on branches increment minor to 1.1.0**_
> on branches increment major to 2.0.0

What I understand from your reply, is that if a branch is configured for an increment (PATCH, MINOR, or MAJOR), the bump message doesn't override the increment configuration for the given branch.  For my example, using the config I provided, the develop branch is configured 'MINOR' increments (increment: Minor). So even though, in the commit message I put +semver: patch. That will never happen.  I was assuming that the bump message gave the user the ability to override the increment configuration for that branch.  So in my example, I was expecting 1.4.1-dev but because the develop branch increment is configure for MINOR, it's ignoring the message bump?

**Reply by HHobeck** [reply]
>So in my example, I was expecting 1.4.1-dev but because the develop branch increment is configure for MINOR, it's ignoring the message bump?

Actually to be precise the algorithm consolidates the increments and uses the highest one. In your case the increment of the branch is higher than the one in the bump message.
